recipe.py

```diff
@@ -34,9 +34,13 @@ def get_ingredients(self):
     
     def set_ingredients(self,ingredients):
         self._ingredients.append(ingredients)
-
+        if len(set(self._ingredients)) != len(self._ingredients):
+            self._ingredients.remove(ingredients) #I have tested this and it only removes the recently added ingredient
+            return False
+    
     def remove_ingredient(self,ingredient):
         self._ingredients.remove(ingredient)
+        #Do I need to prevent this removing duplicates if I have already prevented duplication with the set ingredients validation
```

Comment:
I'm not sure about removing a single ingredient over creating a new modified list of ingredients and simply setting rather than using remove - hmm I will need to have a think.

Reply:
So it is possible that remove_ingredient() will not be needed and we will ask the user to create a new list and set that? Should I remove remove_ingredient()? Would a new function to set ingredients list need to be created?

Reply:
It depends on how the user interface works to a degree, since it's principally at that point one of two things could happen, the UI could create it's own temporary list of changes/additions by the user and then use the temporary list to set a Recipe class or it could simply use functionality within an instance of a Recipe class to do the work. Some might think that the UI should decide when values are correct and should be verified (through application of a save or apply button for example - if you ever had a spell checker that goes berserk before you have even finished the word you can probably see what I mean) others would suggest that you need to check as you are going on as to whether values being entered are correct and as such you need to create the class and apply the changes to see if they are valid - but equally if you are making continuous changes then this carries its own problems like the spelling correction example.

Reply:
ok i will keep in remove ingredient